src/screen.rs

```diff
@@ -1078,10 +1078,30 @@ impl Screen {
         self.need_clear_lines = false;
         self.need_clear_screen = false;
 
+        // Compare the number of lines in the (new) left prompt compared to the
+        // current prompt before it's dropped so that the lines can be cleared
+        // appropriately later (see #11875). Avoid unnecessary clearing by only
+        // doing this when `is_final_rendering`.
+        let mut prompt_has_fewer_lines = is_final_rendering;
+        if is_final_rendering {
+            let actual_prompt_line_breaks = self.actual_left_prompt.as_ref().map_or(1, |p| {
+                let mut result = 1;
```

Comment:
this is inconsistent with the variable name; it ought to be 0 (assuming there is no better solution)